### PR TITLE
[IA-1304] Automation test fix for lockedBy hash 

### PR DESCRIPTION
**Pull request by akarukappadath**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookDataSyncingSpec.scala [hunk 1]
```diff
@@ -59,13 +61,16 @@ class NotebookDataSyncingSpec extends ClusterFixtureSpec with NotebookTestUtils
                 remoteContentSize shouldBe localContentSize
               }
 
+              val hashedUser = Some(String.format("%064x", new BigInteger(1, MessageDigest.getInstance("SHA-256").digest((gcsPath.bucketName + ":" + clusterFixture.cluster.creator).getBytes("UTF-8")))))
```
Comment (jdcanas) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookDataSyncingSpec.scala]: I'd like to see this split into two lines with the `MessageDigest....` as its own val with a meaningful name. I find it a bit hard to read as is.

**Review by jdcanas — APPROVED** [review]
:shipit:

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookDataSyncingSpec.scala [hunk 2]
```diff
@@ -59,13 +61,16 @@ class NotebookDataSyncingSpec extends ClusterFixtureSpec with NotebookTestUtils
...
+
               eventually(timeout(Span(4, Minutes)), interval(Span(30, Seconds))) {
                 val gcsLockedBy: Option[String] = getLockedBy(gcsPath.bucketName, GcsBlobName(gcsPath.objectName.value)).unsafeRunSync()
                 val welderLockedBy: Option[String] = Welder.getMetadata(clusterFixture.cluster, gcsPath, isEditMode).lastLockedBy
 
                 gcsLockedBy should not be None
                 welderLockedBy should not be None
                 gcsLockedBy shouldBe welderLockedBy
+                gcsLockedBy shouldBe hashedUser
```
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookDataSyncingSpec.scala]: what is `welderLockedBy` in the line before this?

Comment (akarukappadath) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookDataSyncingSpec.scala]: After welder updates the `lockedBy` metadata, we're reading the value from the bucket twice - once through welder (`welderLockedBy`) and once directly through gcs (`gcsLockedBy`). This basically checks to make sure that welder can correctly read the `lockedBy` metadata of a notebook, but we weren't checking to see if welder is correctly locking it in the first place, which is what the new check does.

**Review by jdcanas — APPROVED** [review]
👍

File: project/Dependencies.scala [hunk 3]
```diff
@@ -162,6 +162,7 @@ object Dependencies {
     "com.fasterxml.jackson.core" % "jackson-core"         % jacksonV,
     "com.fasterxml.jackson.module" % ("jackson-module-scala_" + scalaV) % jacksonV,
     "ch.qos.logback"  % "logback-classic" % "1.2.3"  % "test",
+    "com.google.guava" % "guava" % "28.0-jre",
```
Comment (rtitle) [inline project/Dependencies.scala:165]: Do we need this? I don't see guava imported.

**Review by rtitle — APPROVED** [review]
LGTM

File: project/Dependencies.scala (hunk 3 above)
Comment (akarukappadath) [reply project/Dependencies.scala:165]: It looks like guava was being brought in by another package - this fixes the error we were seeing when trying to run autotests locally

**Pull request merged**